Lesson 10: Add cart (web & api)

Skurik-SA opened the pull request.

Skurik-SA committed:
Lesson6: Add OAuth

Skurik-SA committed:
Lesson6: Add OAuth (github, vk, google)

Skurik-SA committed:
Lesson 7: Add product categories (web)

Skurik-SA committed:
Lesson 7: Add product categories (web)

Skurik-SA committed:
Lesson 7: Add product categories (web & api)

Skurik-SA committed:
Lesson 7: Add product categories (web & api)

Skurik-SA committed:
Lesson 8: Add products and attributes (web)

Skurik-SA committed:
Lesson 8: Add products and attributes (web)

Skurik-SA committed:
Lesson 8: Add products and attributes (web & api)

Skurik-SA committed:
Lesson 8: Add products and attributes (web & api)

Skurik-SA committed:
minor changes

Skurik-SA committed:
Lesson 9: Add search, sorting, filters to catalog web

Skurik-SA committed:
Lesson 9: Add search, sorting, filters to catalog web

Skurik-SA committed:
Lesson 9: add filters (web and api)

Skurik-SA committed:
Lesson 9: add filters (web and api)

Skurik-SA committed:
some changes

Skurik-SA committed:
some changes

Skurik-SA committed:
some changes

Skurik-SA committed:
Lesson 9: add filters (web and api)

Skurik-SA committed:
Lesson 9: add filters (web and api)

Skurik-SA committed:
Lesson 9: add filters (web and api)

Skurik-SA committed:
minor changes

Skurik-SA committed:
minor changes

Skurik-SA committed:
fixed tags

Skurik-SA committed:
fixed tags

Skurik-SA committed:
Lesson 10: add cart (web and api)

Skurik-SA committed:
Lesson 10: add cart (web and api)

Skurik-SA committed:
Lesson 10: Add cart (web & api)

Skurik-SA committed:
Lesson 10: Add cart (web & api)

chrn-feip commented on fefu-app2/app/Http/Controllers/Api/CartController.php:
```diff
@@ -0,0 +1,62 @@
+     * @param CartModificationRequest $request
+     * @return CartResource
+     */
+    #[OpenApi\Operation(tags: ['cart'], method: 'POST')]
```
Где тело запроса с modifications?

chrn-feip commented on fefu-app2/app/Models/Cart.php:
```diff
@@ -0,0 +1,140 @@
+        foreach ($this->itemsByProductId as $productId => $cartItem) {
+            $cartItem->price_item = $cartItem->product->price;
+            $cartItem->price_total = $cartItem->quantity * $cartItem->price_item;
+            $this->price_total += $cartItem->price_total;
```
Перед циклом надо сбросить значение в 0.

chrn-feip commented on fefu-app2/api-docs.json:
```diff
@@ -277,6 +414,137 @@
             }
         },
+        "\/api\/cart\/set_quantity": {
+            "post": {
```
Где массив modifications, который в методе контроллера валидируется и обрабатывается?

Skurik-SA committed:
fixed

Я убрал из hidden поля. Их не нужно скрывать потому что они всем известны и никому никакой ненужной пользы не принесут?
И добавил время входа и регистрации для обычной авторизации.

Skurik-SA committed:
fixed

Сделал безопасное удаление.

Skurik-SA committed:
fixed

Skurik-SA committed:
Merge branch 'lesson10' of https://github.com/Skurik-SA/fefu_backend_3_6 into lesson10

Skurik-SA committed:
Merge branch 'main' into lesson10

chrn-feip commented:
Сделайте ребейз, пожалуйста.

Skurik-SA commented:
оке

Skurik-SA committed:
Merge branch 'main' into lesson10

Skurik-SA commented:
готово, вроде как
